flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -775,9 +775,14 @@ public final void notifyPartitionDataAvailable(final ResultPartitionID partition
     protected void notifyPartitionDataAvailableInternal(
             IntermediateResultPartitionID resultPartitionId) {}
 
+    /**
+     * Returns a copy of the current history of task failures.
+     *
+     * @return a copy of the current history of task failures.
+     */
     @VisibleForTesting
     protected List<TaskFailureHistoryEntry> getExceptionHistory() {
-        return taskFailureHistory;
+        return new ArrayList<>(taskFailureHistory);
```

Comment:
Why do you need a copy, and why would wrapping it in an unmodifiable list not suffice?